From c42c7ca389e2aa7091bf28ae476f3eb1b3f18a57 Mon Sep 17 00:00:00 2001
From: Dinolek <git@dinolek.me>
Date: Tue, 1 Jul 2025 12:04:42 +0200
Subject: [PATCH 084/115] Input: novatek-nvt-ts - simplify with dev_err_probe()

Error handling in probe can be a bit simpler with dev_err_probe().

Signed-off-by: Dinolek <git@dinolek.me>
---
 .../input/touchscreen/novatek-nvt-ts-core.c   | 33 +++++++------------
 1 file changed, 11 insertions(+), 22 deletions(-)

diff --git a/drivers/input/touchscreen/novatek-nvt-ts-core.c b/drivers/input/touchscreen/novatek-nvt-ts-core.c
index 63a7f88e3..582cefc26 100644
--- a/drivers/input/touchscreen/novatek-nvt-ts-core.c
+++ b/drivers/input/touchscreen/novatek-nvt-ts-core.c
@@ -135,10 +135,8 @@ int nvt_ts_probe(struct device *dev, int irq, struct regmap *regmap,
 	const struct nvt_ts_chip_data *chip;
 	struct input_dev *input;
 
-	if (!irq) {
-		dev_err(dev, "Error no irq specified\n");
-		return -EINVAL;
-	}
+	if (!irq)
+		return dev_err_probe(dev, -EINVAL, "missing irq\n");
 
 	data = devm_kzalloc(dev, sizeof(*data), GFP_KERNEL);
 	if (!data)
@@ -160,23 +158,18 @@ int nvt_ts_probe(struct device *dev, int irq, struct regmap *regmap,
 	data->regulators[0].supply = "vcc";
 	data->regulators[1].supply = "iovcc";
 	error = devm_regulator_bulk_get(dev, ARRAY_SIZE(data->regulators), data->regulators);
-	if (error) {
-		dev_err(dev, "cannot get regulators: %d\n", error);
-		return error;
-	}
+	if (error)
+		return dev_err_probe(dev, error, "cannot get regulators\n");
 
 	error = regulator_bulk_enable(ARRAY_SIZE(data->regulators), data->regulators);
-	if (error) {
-		dev_err(dev, "failed to enable regulators: %d\n", error);
-		return error;
-	}
+	if (error)
+		return dev_err_probe(dev, error, "failed to enable regulators\n");
 
 	data->reset_gpio = devm_gpiod_get(dev, "reset", GPIOD_OUT_LOW);
 	error = PTR_ERR_OR_ZERO(data->reset_gpio);
 	if (error) {
 		regulator_bulk_disable(ARRAY_SIZE(data->regulators), data->regulators);
-		dev_err(dev, "failed to request reset GPIO: %d\n", error);
-		return error;
+		return dev_err_probe(dev, error, "failed to request reset GPIO\n");
 	}
 
 	/* Wait for controller to come out of reset before params read */
@@ -233,16 +226,12 @@ int nvt_ts_probe(struct device *dev, int irq, struct regmap *regmap,
 					  IRQF_ONESHOT | IRQF_NO_AUTOEN |
 						nvt_ts_irq_type[irq_type],
 					  dev_name(dev), data);
-	if (error) {
-		dev_err(dev, "failed to request irq: %d\n", error);
-		return error;
-	}
+	if (error)
+		return dev_err_probe(dev, error, "failed to request irq\n");
 
 	error = input_register_device(input);
-	if (error) {
-		dev_err(dev, "failed to register input device: %d\n", error);
-		return error;
-	}
+	if (error)
+		return dev_err_probe(dev, error, "failed to register input device\n");
 
 	return 0;
 }
-- 
2.56.0

